### ISSUE TITLE
Unabily to refer to other jsonschema

### ISSUE THREAD

**Issue by Thomasator** [joelittlejohn/jsonschema2pojo#1584]
Hello,

I am trying to convert jsonschema to pojo via maven plugin.

I've encountered a problem. My generated jsonschema has references to some other module to other json schema. When I try to generate from json schema to POJO, this references are included in the class file and they are not refered to. For example, I have my own type (enum) 'Boolean', this type is generated into POJO in other module, but when I generate my class, this enum is included at the back of the file. I'd like to refer to the Boolean.java in the other module and not include it in the same file. Is it possible? Also, this file needs to be in the same directory for some reason. I've tried
<sourcePaths>  <sourcePath>external.json</sourcePath>   <sourcePath>this.json</sourcePath>  </sourcePaths>.

Thank u for help
Thomasator

**Comment by abramsz** [comment]
"Boolean" is not a valid type, use "boolean" instead."Boolean" will be treated as a customized type, and lead to creating a new type.

**Comment by Thomasator** [comment]
For example in the jsonschema the reference looks like this "person" : { "$ref" : "external_jsongen.json#/definitions/Person" }, the custum Boolean is needed for my app to work, there are more complex types too.

**Comment by Thomasator** [comment]
I tried some solutions manually, and all it needs is to add include of the java classes for Example 'include com.example.external_module.Person'.

**Comment by joelittlejohn** [comment]
Are you able to edit the schema like this:

```json
{ "existingJavaType" : "com.example.external_module.Person" }
```
?

**Comment by Thomasator** [comment]
Thanks for the response. It works, that is true, but its not the best solution.

I also still need to have the external.json in the same directory. Is there a work around that?

**Comment by Thomasator** [comment]
@joelittlejohn

I've encountred another problem. The solution u mentioned above works for non basic types (enums ...), but if I have "$ref" : "external.json#/definitions/myString" reference to string or int for example, it does not work, because the basic types aren't generated into java classes. Is there some solution for that?

**Comment by unkish** [comment]
> because the basic types aren't generated into java classes. Is there some solution for that?

Not really: code generator(s) can't do anything what you wouldn't be able to do writing code yourself.
For example one can't create class named `int` as latter is a reserved [keyword](https://docs.oracle.com/javase/specs/jls/se17/html/jls-3.html#jls-3.9).

Closing for now, please feel free to reopen if needed.
